include/endian.h

```diff
@@ -146,12 +146,24 @@
 
 #define betoh16               be16toh
 #define letoh16               le16toh
+#define bemtoh16(x)           htobe16(*(uint16_t *)(x))
+#define htobem16(x, v)        (*(uint16_t *)(x) = htobe16(v))
+#define lemtoh16(x)           htole16(*(uint16_t *)(x))
```

Comment:
i guess it's simpler to read if you use letoh16 instead of htole16.
ditto for other macros.